api/solarperformanceinsight_api/compute.py

```diff
@@ -30,22 +35,326 @@ def run_job(job_id: UUID, user: str):
+    else:
+        raise TypeError(f"Unknown result format {type(val)}")
+
+
+def process_single_modelchain(
+    chain: ModelChain,
+    weather_data: List[pd.DataFrame],
+    run_model_method: str,
+    tshift: dt.timedelta,
+    inverter_num: int,
+) -> Tuple[List[DBResult], pd.DataFrame]:
+    """Run and process a single ModelChain
+
+    Parameters
+    ----------
+    chain : pvlib.ModelChain
+        The chain to run
+    weather_data : List[pandas.DataFrame]
+        The data that will be passed to chain.run_model
+    run_model_method : str
+        The method of chain to used calculate results, i.e. run_model or run_from_poa
+    tshift : dt.timedelta
+        Weather data should already be shifted forward by this amount, so final results
+        will be shifted by -1 * tshift.
+    inverter_num : int
+        Which inverter in the full system this chain is for
+
+    Returns
+    -------
+    inverter_results : List[DBResult]
+        List of DBResult objects that can be inserted into the database
+    summary_frame : pd.DataFrame
+        A frame with the AC performance result, zenith angle, and average of
+        poa_global, effective_irradiance, and cell_temperature over all arrays
+    """
+    # run chain
+    mc = getattr(chain, run_model_method)(weather_data)
+    results = mc.results
+    adjust = partial(_adjust_frame, tshift=tshift)
+    performance = adjust(results.ac, "performance")
+    summary_frame = pd.DataFrame(
+        {
+            "poa_global": 0,  # type: ignore
+            "effective_irradiance": 0,  # type: ignore
+            "cell_temperature": 0,  # type: ignore
+        },
+        index=performance.index,
+    )
+    summary_frame.index.name = "time"  # type: ignore
+    # make nice result output
+    num_arrays = len(mc.system.arrays)
+    out = []
+    for i in range(num_arrays):
+        weather: pd.DataFrame = adjust(  # type: ignore
+            _get_index(results, "total_irrad", i)[["poa_global"]], "poa_global"
+        )
+        weather.loc[:, "effective_irradiance"] = adjust(  # type: ignore
+            _get_index(results, "effective_irradiance", i), "effective_irradiance"
+        )
+        weather.loc[:, "cell_temperature"] = adjust(  # type: ignore
+            _get_index(results, "cell_temperature", i), "cell_temperature"
+        )
+        summary_frame += weather  # type: ignore
+        out.append(
+            DBResult(
+                schema_path=f"/inverters/{inverter_num}/arrays/{i}",
+                type="weather data",
+                data=weather,
+            )
+        )
+    out.append(
+        DBResult(
+            schema_path=f"/inverters/{inverter_num}",
+            type="performance data",
+            data=performance,
+        )
+    )
+    # mean
+    summary_frame /= num_arrays  # type: ignore
+    summary_frame.insert(0, "performance", performance)  # type: ignore
+    summary_frame.insert(
+        len(summary_frame.columns),
+        "zenith",
```

Comment:
Should we use apparent zenith? I think that would be more correct for this application, but not much difference at 87 degrees so I'm fine if we want to keep zenith for simplicity.